strawberry/contrib/django/tests/test_model_type.py

```diff
@@ -0,0 +1,167 @@
+            modelType {
+                name
+                extraInfo
+            }
+        }
+    """,
+        schema=schema,
+    )
+
+    assert response.data["modelType"]["name"] == "modeltype"
+    assert response.data["modelType"]["extraInfo"] == "more info"
+
+
+@pytest.mark.asyncio
+async def test_model_type_doesnt_override_type_fields():
+    @model_type(model=TestModel, only_fields=["name"])
+    class TestModelType:
+        @strawberry.field
+        def name(self, info) -> str:
+            return "joe"
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def model_type(root, info) -> TestModelType:
+            return TestModel(name="modeltype")
+
+    schema = strawberry.Schema(query=Query)
+
+    response = await execute(
+        query="""
+        query {
+            modelType {
+                name
+            }
+        }
+    """,
+        schema=schema,
+    )
+
+    assert response.data["modelType"]["name"] == "joe"
+
+
+@pytest.mark.asyncio
+async def test_model_type_exclude_fields():
+    @model_type(model=TestModel, exclude_fields=["secret"])
+    class TestModelType:
+        pass
+
+    assert "secret" not in TestModelType.field.fields
+
+
+@pytest.mark.asyncio
+async def test_model_type_only_fields():
+    @model_type(model=TestModel, only_fields=["name"])
+    class TestModelType:
+        pass
+
+    assert "name" in TestModelType.field.fields
+    assert "id" not in TestModelType.field.fields
+    assert "secret" not in TestModelType.field.fields
+
+
+@pytest.mark.asyncio
+@pytest.mark.django_db
+async def test_model_type_list():
+    @model_type(model=TestModel)
+    class TestModelType:
+        @strawberry.field
+        def extra_info(self, info) -> str:
+            return "more info"
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def model_type_list(root, info) -> List[TestModelType]:
+            return TestModel.objects.all()
+
+    schema = strawberry.Schema(query=Query)
+
+    TestModel.objects.create(name="modeltype")
+    TestModel.objects.create(name="modeltype")
+    TestModel.objects.create(name="modeltype")
+
+    response = await execute(
+        query="""
+        query {
+            modelTypeList {
+                id
+                name
+                extraInfo
+            }
+        }
+    """,
+        schema=schema,
+    )
+
+    assert len(response.data["modelTypeList"]) == 3
+
+
+@pytest.mark.asyncio
+@pytest.mark.django_db
+async def test_model_input_type():
+    @model_type(model=TestModel)
+    class TestModelType:
+        extra: str
+
+    @model_type(model=TestModel, is_input=True)
+    class TestModelInputType:
+        pass
+
+    @strawberry.type
+    class Query:
+        hello: str = "world"
+
+    @strawberry.type
+    class Mutation:
+        @strawberry.field
+        def create_test_model(self, info, test: TestModelInputType) -> TestModelType:
+            return TestModel.objects.create(**test.__dict__)
```

Comment:
uhm, I'm not really convinced by this, doesn't seem user friendly.

I'll think of some alternatives, maybe @marcoacierno has a suggestion.

I guess we also need to come up with a solution for FK and M2M fields (but that can be in another PR)

Reply:
Can you expand a bit on what isn't convincing about this?

Using `**test.__dict__` is just a shorthand for the tests right now. I wouldn't recommend passing data directly from an input type to `objects.create` like this in most cases, since it will only work with some field types. I can change this test to be more verbose if you'd like.

The goal of this test is to show that a mutation resolver can access data from an input `model_type` correctly. As of right now, this PR only support generation of type classes and their usage in resolver definitions. This is definitely not as user friendly as I would like, since most mutation resolvers defined by users will end up have a lot of boilerplate code. I am planning on opening another PR later on to add query / mutation generation for model types (a la DRFs `ModelViewSet `), but that's a little ways off right now. 

We should definitely come up with an automatic solution for FK and M2M fields, but right now that can be achieved with custom resolvers on the `model_type` decorated class. I'll add some tests to show that functionality.

Reply:
>  I am planning on opening another PR later on to add query / mutation generation for model types (a la DRFs ModelViewSet ), but that's a little ways off right now.

Oh awesome!

> Using **test.__dict__ is just a shorthand for the tests right now. I wouldn't recommend passing data directly from an input type to objects.create like this in most cases, since it will only work with some field types. I can change this test to be more verbose if you'd like.

Yes please! Sometime people use tests to see how to achieve something they are doing, so better to show the best practises ;)

Reply:
Sounds good! I'll update the tests :)

Reply:
Done :)